app/controllers/users_controller.rb

```diff
@@ -44,12 +57,23 @@ def add_contact
     redirect_to root_path
   end
 
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/MethodLength
   def search
-    @users = User.search(params[:search]).where.not(id: current_user.id)
-    @users_to_add = @users.reject do |user|
-      current_user.sent_contact_request?(user)
+    use_wildcards = true
+    @users = Set[]
+    @contacts = Set[]
+    if current_user && params[:search]
+      @user = User.find(current_user.id)
+      @users = search_record(params[:search], User, use_wildcards)
```

Comment:
why are you not using the method from the Users model anymore? I don't think this is a good way to implement searching since you are using a method from the users helper but nevertheless have to tell which record you want to search in.
`User.search(params[:search]).where.not(id: current_user.id)` this line gave you all users which match the search parameter and which are not the current user

Reply:
I would strongly support this. The search logic built by Scrumbledore's Army is reviewed and well-tested. I do not see any reason to rebuild this logic. And I would also prefer the approach of the existing solution over your new approach.

Reply:
Hm... You are right. I don't know what happened there 🤔 . I will use the search logic from before.